lib/rucio/core/rule.py

```diff
@@ -1943,7 +2266,14 @@ def update_rules_for_lost_replica(scope, name, rse_id, nowait=False, *, session:
         elif rule.locks_replicating_cnt == 0 and rule.locks_stuck_cnt == 0:
             rule.state = RuleState.OK
             if rule.grouping != RuleGrouping.NONE:
-                session.query(models.DatasetLock).filter_by(rule_id=rule.id).update({'state': LockState.OK})
+                stmt = update(
+                    models.DatasetLock
+                ).where(
+                    models.DatasetLock.rule_id == rule.id
+                ).values(
+                    state=LockState.OK
+                )
```

Comment:
I think it would be preferable if we only used dictionaries instead of keyword parameter values, for the same reason as one of my initial comments. So this should be:

```python
.values({
    models.DatasetLock.state: LockState.OK
})
```

This comment applies throughout the file.

Reply:
All `values()` now use the style you mentioned, i.e.:
`values({
                        models.DatasetLock.state: LockState.STUCK
                    })`